app/components/TrackCard/index.js

```diff
@@ -11,35 +11,71 @@ import { Card } from '@mui/material';
 import T from '@components/T';
 import If from '@components/If';
 import isEmpty from 'lodash/isEmpty';
+import { useLocation } from 'react-router-dom';
 
 const CustomCard = styled(Card)`
   && {
     margin: 1rem 0;
     padding: 1rem;
+    min-height: 270px;
   }
 `;
 
-export function TrackCard({ collectionName, artistName, collectionPrice, artworkUrl100 }) {
-  return (
-    <CustomCard data-testid="repo-card">
-      
-      <If condition={!isEmpty(collectionName)} otherwise={<T data-testid="name-unavailable" id="repo_name_unavailable" />}>
-        <div ><img src={artworkUrl100} alt={collectionName} />
-        <T data-testid="name" id="track_name" values={{ trackName: collectionName }} />
-        </div>
-       
-      </If>
-      <If
-        condition={!isEmpty(artistName)}
-        otherwise={<T data-testid="fullName-unavailable" id="repo_full_name_unavailable" />}
-      >
-        <T data-testid="fullName" id="artist_name" values={{ artist_name: artistName }} />
-      </If>
-      <If condition={collectionPrice} otherwise={<T data-testid="stargazers-unavaiable" id="repo_stars_unavailable" />}>
-        <T data-testid="stargazers" id="total_cost" values={{ price: collectionPrice }} />
-      </If>
-    </CustomCard>
-  );
+export function TrackCard({ collectionName, artistName, collectionPrice, artworkUrl100, description }) {
+  const location = useLocation();
+  console.log(location.state);
+  if(isEmpty(collectionName) && !isEmpty(location.state)){
+    collectionName=location.state.item.collectionName;
+    artistName=location.state.item.artistName;
+    collectionPrice=location.state.item.collectionPrice;
+    artworkUrl100=location.state.item.artworkUrl100;
+    description=location.state.item.description;
+
+    return (
+      <CustomCard data-testid="repo-card">
+        
+        <If condition={!isEmpty(collectionName)} otherwise={<T data-testid="name-unavailable" id="repo_name_unavailable" />}>
+          <div ><img src={artworkUrl100} alt={collectionName} />
+          <T data-testid="name" id="track_name" values={{ trackName: collectionName }} />
+          </div>
+         
+        </If>
+        <If
+          condition={!isEmpty(artistName)}
+          otherwise={<T data-testid="fullName-unavailable" id="repo_full_name_unavailable" />}
+        >
+          <T data-testid="fullName" id="artist_name" values={{ artist_name: artistName }} />
+        </If>
+        <div>Description: {description}</div>
+        <If condition={collectionPrice} otherwise={<T data-testid="stargazers-unavaiable" id="repo_stars_unavailable" />}>
+          <T data-testid="stargazers" id="total_cost" values={{ price: collectionPrice }} />
+        </If>
+      </CustomCard>
+    );
+  }
+  else{
+    return (
+      <CustomCard data-testid="repo-card">
+        
+        <If condition={!isEmpty(collectionName)} otherwise={<T data-testid="name-unavailable" id="repo_name_unavailable" />}>
```

Comment:
At the top level if condition we are checking whether the collection name is empty or not. Then why are we again doing a check here?

Reply:
Here the conditions were put because while routing using history.push the props value are not getting passed directly so I am fetching those values from the global states this is done for "/tracks/:trackid"